substrate/client/network/src/service.rs

```diff
@@ -260,12 +291,17 @@ where
 					.saturating_add(10)
 			};
 
-			transport::build_transport(
-				local_identity.clone(),
-				config_mem,
-				network_config.yamux_window_size,
-				yamux_maximum_buffer_size,
+			transport_builder(
+				NetworkConfig {
+					keypair: local_identity.clone(),
+					memory_only: config_mem,
+					muxer_window_size: network_config.yamux_window_size,
+					muxer_maximum_buffer_size: yamux_maximum_buffer_size,
+				}
 			)
+				.map(|(peer_id, stream_muxer), _| (peer_id, StreamMuxerBox::new(stream_muxer)))
+				.boxed()
+				.with_bandwidth_logging()
```

Comment:
It's not deprecated in this version ([link](https://github.com/libp2p/rust-libp2p/blob/c04a4a1ceaa241f446499cd421a6cdd63e13f23d/libp2p/src/transport_ext.rs#L69)) ;) I would keep it as close as possible to substrate's version - it will be easier to maintain this way.